src/ucp/stream/stream_recv.c

```diff
@@ -30,110 +62,311 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_stream_recv_data_nb,
 
     UCP_THREAD_CS_ENTER_CONDITIONAL(&ep->worker->mt_lock);
 
-    if (ucs_queue_is_empty(&ep->ext.stream->data)) {
+    rdesc = ucp_stream_recv_data_nb_internal(ep->ext.stream);
+    if (ucs_likely(rdesc != NULL)) {
+        *length = rdesc->length;
+        ret = ucp_stream_rdesc_data(rdesc);
+    } else {
```

Comment:
in 'else' case, can just do ret=NULL.

Reply:
it causes compilation error with -Werror,-Wnon-literal-null-conversion, restored.